### PR TITLE
Use `__` as a prefix for compiler intrinsics 

### LINKED ISSUE #1531: Domain-separate intrinsics, i.e. <code class="notranslate">__size_of()` and <code class="notranslate">__is_reference_type()`.

### LINKED ISSUE #1533: Forbid double leading underscores in function, variable, struct, enum, etc. names.

### PR DESCRIPTION
**Pull request by sezna**
Closes #1531
Closes #1533

### CODE REVIEW

**Review by adlerjohn — COMMENTED** [review]
CI failing

**Comment by mohammadfawaz** [comment]
I think you just forgot to update "should_pass/language/size_of"

**Comment by adlerjohn** [comment]
> Yeah but your `size_of` is not `size_of`

**Review by adlerjohn — COMMENTED** [review]
fmt Sway files, and it looks like none of the added tests actually use `core`, so use `implicit_std = false` instead.

File: sway-parse/src/error.rs [hunk 1]
```diff
@@ -2,66 +2,68 @@ use crate::priv_prelude::*;
 
 #[derive(Debug, Error, Clone, PartialEq, Hash)]
 pub enum ParseErrorKind {
-    #[error("expected an import name, group of imports, or `*`")]
+    #[error("Expected an import name, group of imports, or `*`.")]
     ExpectedImportNameGroupOrGlob,
-    #[error("expected an item")]
+    #[error("Expected an item")]
```
Comment (otrho) [inline sway-parse/src/error.rs:7]:
```suggestion
    #[error("Expected an item.")]
```

File: sway-parse/src/item/mod.rs [hunk 2]
```diff
@@ -257,7 +257,7 @@ impl Parse for FnSignature {
         let visibility = parser.take();
         let impure = parser.take();
         let fn_token = parser.parse()?;
-        let name = parser.parse()?;
+        let name: Ident = parser.parse()?;
```
Comment (otrho) [inline sway-parse/src/item/mod.rs:260]: Why is this change needed all of a sudden?  Isn't the type implied on line 285?

**Review by otrho — CHANGES_REQUESTED** [review]
Looks good otherwise.

**Review by adlerjohn — CHANGES_REQUESTED** [review]

**Review by adlerjohn — APPROVED** [review]

**Review by mohammadfawaz — APPROVED** [review]

**Review by otrho — APPROVED** [review]

**Pull request merged**